Added AWS tutorial (#339)
haiku/website: content/guides/virtualizing/aws.md

* [Preparing the AMI](#part_ami)
* [Deploying the AMI on the AWS platform](#part_deployment)

### Preparing for this tutorial <a name="part_prepare">

Disk files represent physical devices but in the form of a file. During the preparation phase, you will need to download a Haiku image file from [here](https://www.haiku-os.org/get-haiku). Then, you will have to use [VirtualBox](https://www.virtualbox.org/) in order to create a disk image that can be later converted into an AMI image file.

If you're not familiar with this process, then you can use the **[Virtualizing Haiku in VirtualBox](https://www.haiku-os.org/guides/virtualizing/virtualbox/)** tutorial as a reference.

From the perspective of the operating system, disk images can be compared to physical hard drives and other forms of storage. Therefore, it should be noted that while creating your virtual machine in VirtualBox, you should ensure that the disk space that was allocated to the drive does not cross your quota on the AWS platform in order to avoid any unnecessary charges.

### Preparing the virtual disk image <a name="part_prepare">

##### Spotting the virtual disk image

> After following through the process, you should now have a `.vdi` file alongside other relevant. Its precise location can vary according to your operating system, but generally speaking;

Author: Accidentally deleted "relevant **files**"

```
vboxmanage clonehd <source path of the .vdh file> <destination path of the .vhd file> --format VHD
```

### Preparing the environment <a name="part_environment">

Now that we have successfully extracted the `.vdi` file, we will need to convert it into an AMI image. Fortunately for us, Amazon provides tools that are designed for this operation.

- In Debian/Ubuntu machines, you can obtain them by running the following command inside of a Terminal:

```
sudo apt install ec2-api-tools ec2-ami-tools
```

- In Linux-based operating systems that use the [Red Hat Package Manager](https://rpm.org) (RPM), such as CentOS, Fedora, Red Hat Linux, you can obtain the tools by running these commands instead:

```
sudo yum install ruby

# Install ec2-api tools
wget http://s3.amazonaws.com/ec2-downloads/ec2-api-tools.zip
unzip ec2-api-tools.zip
cd ./ec2-api-tools-*
mkdir /opt/ec2 # feel free to use other directories
mv ./* /opt/ec2/tools

# Install ec2-ami-tools
wget https://s3.amazonaws.com/ec2-downloads/ec2-ami-tools.noarch.rpm
sudo yum install ec2-ami-tools.noarch.rpm
```

> Make sure to fill out the information in between of the `<>`'s.

Reviewer: There are no <> in the above commands.

We're nearly done! All we need is to store our credentials in variables, since we will use them multiple times during the next section of the tutorial.

For convenience, we will set our [AWS Security Credentials](https://docs.aws.amazon.com/general/latest/gr/aws-security-credentials.html) as two variables; `AWS_KEY` and `AWS_SEC`.

```
export AWS_KEY="<YOUR AWS KEY>"
export AWS_SEC="<YOUR AWS SECRET>"
```

> Once again, make sure to replace the two placeholders in between of the `<>`'s with your credentials.

Reviewer: Not "Once again" actually.
As described, it'd mean to keep the pointy brackets. If that's not correct, maybe be totally explicit with:
> Make sure to replace `<YOUR AWS KEY>` and `<YOUR AWS SCRET>` with your actual AWS key and secret.

### Preparing the AMI <a name="part_ami">

##### Uploading the disk image

> Now that you have gotten everything set and in order, we can now upload the image over to Amazon!

Reviewer: In this guide you decided to mostly used "we now do this", "we now do that". We should avoid switching to "you now do this", "you now do that". So:
> Now we got everything set and in order, we upload the image to Amazon!

```
ec2-import-volume \
    --format vhd \
    --volume-size 4 \
    --region us-east-1 \
    --availability-zone us-east-1a \
    --bucket haiku-folder \
    --owner-akid $AWS_KEY --owner-sak $AWS_SEC \
    --aws-access-key $AWS_KEY --aws-secret-key $AWS_SEC \
    Haiku.vhd
```

Before proceeding any further, we will go over some of the parameters which you may find necessary to change.

- `--volume-size` is the parameter that defines the amount of space that will be allocated for the virtualized environment. In this case, we chose the number `4` because it is the same one as the one we picked during the installation process in the [VirtualBox tutorial](https://www.haiku-os.org/guides/virtualizing/virtualbox/).

> - `--region` is the parameter that sets the server's region, which is independent from other regions. The `--availability-zone` parameter sets the availability zone-- they also hold a degree of independence, but they're a part of something that could be compared to an interlinked cluster. You can learn more about this topic in the [AWS User Guide](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html)

Reviewer: ```suggestion
- `--region` is the parameter that sets the server's region, which is independent from other regions. The `--availability-zone` parameter sets the availability zone -- those also hold a degree of independence, but are a part of something that could be compared to an interlinked cluster. You can learn more about this topic in the [AWS User Guide](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html)
```

- `--bucket` defines the name of your Amazon S3 Bucket. You can find out more about about buckets [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html).

##### Creating the AMI

> Now that we have uploaded our image, we can now proceed with making a snapshot.

Reviewer: ```suggestion
Now that we have uploaded our image, we can now proceed with making a snapshot:
```

```
ec2-create-snapshot \
   --aws-access-key $AWS_KEY \
   --aws-secret-key $AWS_SEC \
   --region us-east-1 \
   haiku-folder
```

> Snapshots are particularly useful in our case, because we can make an AMI out of it. 

Reviewer: ```suggestion
Snapshots are particularly useful in our case, because we can make an AMI out of it:
```

```
ec2-register \
   --name "Haiku R1/beta1 AMI" \
   --aws-access-key $AWS_KEY \
   --aws-secret-key $AWS_SEC \
   --region us-east-1 \
   --architecture x86_64 \
   --root-device-name /dev/sda1 \
   --virtualization-type hvm \
   --snapshot haiku
```

### Deploying the AMI on the AWS platform <a name="part_deployment">

> After creating the AMI, it will appear in the `AMIs` section of the AWS Explorer. You can now launch an instance of the AMI by clicking on the `My AMIs` tab and selecting your newly made Haiku AMI.

Reviewer: Again we <-> you. Maybe go through the guide again and change to "we/our" where appropriate.

```suggestion
After creating the AMI, it will appear in the `AMIs` section of the AWS Explorer. We can now launch an instance of the AMI by clicking on the `My AMIs` tab and selecting our newly made Haiku AMI.
```

##### Creating the AMI

Now that we have uploaded our image, we can now proceed with making a snapshot:

```sh

> ec2-create-snapshot \

Reviewer 2: Perhaps this was a typo? But for creating a snapshot with this command, it requires the `Volume ID` from the Elastic Block Store in EC2 after importing the volume rather than the bucket name.

Replace `haiku-folder` with `vol-XXXXXXXXXX` where the latter is the volume is used for the snapshot.
